lib/modules/datasource/go/releases-goproxy.ts

```diff
@@ -52,21 +52,38 @@ export class GoProxyDatasource extends Datasource {
           break;
         }
 
-        const releasesIndex = await this.listVersions(url, packageName);
-        const releases = await p.map(releasesIndex, async (versionInfo) => {
-          const { version, releaseTimestamp } = versionInfo;
-
-          if (releaseTimestamp) {
-            return { version, releaseTimestamp };
+        let releases: Release[] = [];
+        if (packageName.startsWith('gopkg.in/')) {
+          const versionSuffixRegex = regEx(/\.v(\d+)$/);
+          const versionSuffixMatch = packageName.match(versionSuffixRegex);
+          if (versionSuffixMatch === null || versionSuffixMatch.length < 2) {
+            releases = await this.getVersionsWithInfo(url, packageName);
+          } else {
+            const versionSuffix = parseInt(versionSuffixMatch[1]);
+            for (let i = 0; ; i++) {
+              try {
+                releases = releases.concat(
+                  await this.getVersionsWithInfo(
+                    url,
+                    packageName.replace(
+                      versionSuffixRegex,
+                      `.v${versionSuffix + i}`
+                    )
+                  )
+                );
+              } catch (err) {
+                const statusCode = err?.response?.statusCode;
+                if (i > 0 && (statusCode === 404 || statusCode === 410)) {
+                  break;
+                }
```

Comment:
These are the *expected* error codes if the major release doesn't exist?

Reply:
Yes, catching them avoids to fallback to another proxy, if the current major was available, I think we can expect the following would be too

https://github.com/renovatebot/renovate/blob/21ee796d6bd718d208ae0c862bf96106478eb79a/lib/modules/datasource/go/releases-goproxy.ts#L86-L87